src/test/index.ts

```diff
@@ -10,6 +10,10 @@ require('./common/exitCIAfterTestReporter');
 if ((Reflect as any).metadata === undefined) {
     require('reflect-metadata');
 }
+// Always place at top, must be done before we import any of the files from src/client folder.
+// We need to ensure nyc gets a change to setup necessary hooks before files are loaded.
+const { setupCoverage } = require('./coverage');
+const nyc = setupCoverage();
```

Comment:
Using `raw` require instead of `import` to ensure this is on top just as we do with `reflect-metadata`.